conan_provider.cmake

```diff
@@ -229,6 +275,10 @@ macro(conan_provide_dependency package_name)
     endif()
 
     get_property(CONAN_GENERATORS_FOLDER GLOBAL PROPERTY CONAN_GENERATORS_FOLDER)
+    list(FIND CMAKE_FIND_ROOT_PATH "${CONAN_GENERATORS_FOLDER}" index)
```

Comment:
This probably deserves a comment here, saying why this is neessary.

Reply:
mmm I'm not entirely sure about this one, is this 100% needed?

The `CONAN_GENERATORS_FOLDER` would only contain files generated by `CMakeDeps` to resolve `find_package` calls.  Ensuring that `CMAKE_FIND_ROOT_PATH` needs to be looked at first, would only be needed if loading the Android toolchain causes CMake to look for packages there _first_, possibly finding dependencies outside of the intended Conan packages.

However, does that actually happen? I cannot find any `.cmake` finds inside the android NDK that would satisfy any `find_package` call - at least not in NDK version r25b which is the one I'm looking at.

If this is indeed needed because there are scenarios in which the wrong package is found - I would probably suggest taking advantage of the fact that `find_package` is relegated to the dependency provider - thus we can alter the search order when invoking `find_package` rather than rely on global flags

Reply:
When I don't have `CMAKE_FIND_ROOT_PATH` set, I get this error in the example:

```
CMake Error at /Users/steve/Code/cmake-conan/conan_provider.cmake:291 (find_package):
  By not providing "Findfmt.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "fmt", but
  CMake did not find one.
```

So it's not that it's finding the wrong package, it's just that it's not finding anything for some reason. This is consistent with my experiences in other codebases as well, but only for cross-compiling. I can look into this in a separate PR, I've long wondered why `CMAKE_PREFIX_PATH` is not enough!

Reply:
When temporarily adding `set(CMAKE_FIND_DEBUG_MODE TRUE)` to the example `CMakeLists.txt` before `find_package()`, it shows all the paths it searches. When the Android toolchain is used, it's not even trying to use the `CMAKE_PREFIX_PATH` locations. It just searches inside the Android NDK itself. It's as-if `set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)` is set, but I don't see it in the regular toolchain, only the legacy one as you observed. Perhaps 'ONLY' is the default behavior of a toolchain? Anyway, I stopped setting the `CMAKE_FIND_ROOT_PATH` variable and instead set `CMAKE_FIND_ROOT_PATH_BOTH` inside `find_package()`. Let me know what you think.